tests/pipeline/utils.py

```diff
@@ -268,36 +298,89 @@ def load_tables_to_dicts(
+        p (dlt.Pipeline): Pipeline whose destination should be accessed.
+        *table_names (str): Optional list of table names to load. If omitted all
+            data tables from the pipeline's default schema are selected.
+        schema_name (str, optional): Explicit schema override. Defaults to the
+            pipeline's default schema.
+        exclude_system_cols (bool, optional): When ``True`` remove columns that
+            start with the ``_dlt`` prefix from the returned data. Defaults to
+            ``False``.
+        sortkey (str, optional): Name of a column to lexicographically sort the
+            resulting list of dictionaries inside each table.
+
+    Returns:
+        Dict[str, List[Dict[str, Any]]]: Table data keyed by table name.
+    """
+
+    if not table_names:
+        table_names = [t["name"] for t in p.default_schema.data_tables()]  # type: ignore[assignment]
+
+    # filesystem with sftp requires a fallback
+    if _is_sftp(p):
+        result = _load_tables_to_dicts_fs(p, *table_names, schema_name=schema_name)
+    else:
+        result = _load_tables_to_dicts_sql(p, *table_names, schema_name=schema_name)
+
+    # exclude and sort
     def _exclude_system_cols(dict_: Dict[str, Any]) -> Dict[str, Any]:
         return {k: v for k, v in dict_.items() if not k.startswith("_dlt")}
 
     def _sort_list_of_dicts(list_: List[Dict[str, Any]], sortkey: str) -> List[Dict[str, Any]]:
         """Sort list of dictionaries by dictionary key."""
         return sorted(list_, key=lambda d: d[sortkey])
 
-    if _is_filesystem(p):
-        result = _load_tables_to_dicts_fs(p, *table_names, schema_name=schema_name)
-    else:
-        result = _load_tables_to_dicts_sql(p, *table_names, schema_name=schema_name)
-
     if exclude_system_cols:
         result = {k: [_exclude_system_cols(d) for d in v] for k, v in result.items()}
     if sortkey is not None:
         result = {k: _sort_list_of_dicts(v, sortkey) for k, v in result.items()}
+
+    # done
     return result
 
 
 def assert_records_as_set(actual: List[Dict[str, Any]], expected: List[Dict[str, Any]]) -> None:
-    """Compares two lists of dicts regardless of order"""
-    actual_set = set(frozenset(dict_.items()) for dict_ in actual)
-    expected_set = set(frozenset(dict_.items()) for dict_ in expected)
-    assert actual_set == expected_set
+    """Assert that *actual* and *expected* contain the same records irrespective of order.
+
+    Args:
+        actual (List[Dict[str, Any]]): Records retrieved from the destination.
+        expected (List[Dict[str, Any]]): Reference records defined in the test.
+
+    Raises:
+        AssertionError: When the two sets of records differ.
+    """
+
+    def dict_to_tuple(d):
+        # Sort items to ensure consistent ordering
+        return tuple(sorted(d.items()))
+
+    counter1 = Counter(dict_to_tuple(d) for d in actual)
+    counter2 = Counter(dict_to_tuple(d) for d in expected)
+
+    assert counter1 == counter2
```

Comment:
praise: nice improvement, faster, better error-msg I suppose and checks row-counts too